src/symbolic/symbolic_path_condition.ml

```diff
@@ -32,7 +32,9 @@ let to_set pc =
 (* Return the set of constraints from [pc] that are relevant for [c]. *)
 let slice (pc : t) (c : Symbolic_value.vbool) : Smtml.Expr.Set.t =
   match Smtml.Expr.get_symbols [ c ] with
-  | [] -> Smtml.Expr.Set.singleton c
+  | [] ->
+    (* TODO: using Smtml.Expr.Set.empty seems to be working but it looks suspicious, it should be tested *)
+    Smtml.Expr.Set.singleton c
```

Comment:
I guess if `get_symbols` returns `[]` it must be a concrete expression that is not being simplified properly :cry:

Reply:
Hum... I believe this branch is sometimes reached. I can provide the list of expressions that are not simplified if you're interested? :)

Reply:
I added a commit that fails on such not properly simplified expressions, see https://github.com/OCamlPro/owi/pull/494/commits/5d4b35b69c2ac8ef83fc8dfef982ba8b955db492

They all seem to be related to `++`, `extract` and `extend`.

Reply:
Ah thanks! I think there are a few of these that I can fix easily.

The ones with pointers are more tricky because I don't want to simplify pointers so much that we loose the `Ptr` type from a given expression. Then, we would loose the ability to track heap chunks. 

Do you think having something like an `unsafe_simplify`, that is more aggressive and simplifies every concrete expression to a given value helpful here?

Reply:
I need to think more about it to get a clear idea of what we want. I believe you know better than me so, `unsafe_simplify` is probably OK for now! :)